features/prog.go

```diff
@@ -248,7 +252,7 @@ func haveProgramHelper(pt ebpf.ProgramType, helper asm.BuiltinFunc) error {
 
 func progLoadProbeNotImplemented(pt ebpf.ProgramType) bool {
 	switch pt {
-	case ebpf.Tracing, ebpf.StructOps, ebpf.Extension, ebpf.LSM:
```

Comment:
We might want to consider the StructOps program probe separately. This works fine for only probing for the program type but will allow using it in the HaveProgramHelper API as well which probably won't work: 
assuming e.g. `HaveProgramHelper(ebpf.StructOps, asm.FnExampleHelper)` will return ENOTSUPP as with the regular `HaveProgramType` feature test, but won't actually tell us anything about the StructOps/Helper combination.

Reply:
We could allow it for `HaveProgramType` but not for `HaveProgramHelper`.

Reply:
Yep, this approach won't work for probing helper types. libbpf hasn't implemented this yet, either: https://elixir.bootlin.com/linux/v5.18.13/source/tools/lib/bpf/libbpf_probes.c#L403.

Crafting a correct program could be a follow-up goal, and would require finding a correct type ID in vmlinux. (I assume @qmonnet's patch suggestion in #344 successfully created a map?) From the looks of it, BTF marshaling might be necessary for that.

Reply:
@rgo3 I think I see what you mean. Removing `ebpf.StructOps` from this switch case here will make `HaveProgramHelper(ebpf.StructOps,  ...)` calls now return `ENOTSUPP` instead of the more user-friendly `"a probe for ProgType ebpf.StructOps isn't implemented"`. I think taking this small hit is acceptable if it means we don't need to refactor this.